mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m

```diff
@@ -449,25 +452,29 @@ - (IBAction)useVerboseLoggingCheckBoxAction:(id)sender {
 }
 
 - (void)handleRequestToInstallPackage:(KMPackage *) package {
+  os_log_debug([KMLogs dataLog], "handleRequestToInstallPackage");
   NSString *keyboardInfoString = NSLocalizedString(@"info-install-keyboard-filename", nil);
   [self.confirmKmpInstallAlertView setInformativeText:[NSString localizedStringWithFormat:keyboardInfoString, package.getOrigKmpFilename]];
   
   os_log_debug([KMLogs uiLog], "Asking user to confirm installation of %{public}@, KMP - temp file name: %{public}@", package.getOrigKmpFilename, package.getTempKmpFilename);
   
-  [self.confirmKmpInstallAlertView beginSheetModalForWindow:self.window
-                                              modalDelegate:self
-                                             didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:)
-                                                contextInfo:(__bridge void *)(package)];
+  [self.confirmKmpInstallAlertView beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse returnCode)  {
+    os_log_debug([KMLogs uiLog], "confirm keyboard installation alert dismissed");
+    if (returnCode == NSAlertFirstButtonReturn) {
+      [self installPackageFile: package.getTempKmpFilename];
+      self.confirmKmpInstallAlertView = nil;
```

Comment:
Does this mean that `self.confirmKmpInstallAlertView` is not nilled if another returnCode is given?

Reply:
Yes, but it should be. I didn't move that over correctly when I created the new handler. I've changed it now.